bindings/rust/s2n-tls/src/cert_chain.rs

```diff
@@ -75,20 +107,115 @@ impl CertificateChain<'_> {
     pub(crate) fn as_mut_ptr(&mut self) -> NonNull<s2n_cert_chain_and_key> {
         self.ptr
     }
+
+    /// Retrieve a reference to the [`CertificateChainContext`] stored on the CertificateChain.
+    pub(crate) fn context(&self) -> &CertificateChainContext {
+        unsafe {
+            let ctx = s2n_cert_chain_and_key_get_ctx(self.ptr.as_ptr())
+                .into_result()
+                .unwrap();
+            &*(ctx.as_ptr() as *const CertificateChainContext)
+        }
+    }
+
+    /// Retrieve a mutable reference to the [`CertificateChainContext`] stored on the CertificateChain.
+    pub(crate) fn context_mut(&mut self) -> &mut CertificateChainContext {
+        unsafe {
+            let ctx = s2n_cert_chain_and_key_get_ctx(self.ptr.as_ptr())
+                .into_result()
+                .unwrap();
+            &mut *(ctx.as_ptr() as *mut CertificateChainContext)
+        }
+    }
+
+    pub fn load_pem(&mut self, certificate: &[u8], private_key: &[u8]) -> Result<&mut Self, Error> {
+        let certificate = CString::new(certificate).map_err(|_| Error::INVALID_INPUT)?;
+        let private_key = CString::new(private_key).map_err(|_| Error::INVALID_INPUT)?;
+        unsafe {
+            s2n_cert_chain_and_key_load_pem(
+                self.ptr.as_ptr(),
+                certificate.as_ptr(),
+                private_key.as_ptr(),
+            )
+            .into_result()
+        }?;
+        Ok(self)
+    }
+
+    pub fn set_ocsp_data(&mut self, data: &[u8]) -> Result<&mut Self, Error> {
+        let size: u32 = data.len().try_into().map_err(|_| Error::INVALID_INPUT)?;
+        unsafe {
+            s2n_cert_chain_and_key_set_ocsp_data(self.ptr.as_ptr(), data.as_ptr(), size)
+                .into_result()
+        }?;
+        Ok(self)
+    }
+}
+
+impl Clone for CertificateChain<'_> {
+    fn clone(&self) -> Self {
+        let context = self.context();
+
+        // Safety
+        //
+        // Using a relaxed ordering is alright here, as knowledge of the
+        // original reference prevents other threads from erroneously deleting
+        // the object.
+        // https://github.com/rust-lang/rust/blob/e012a191d768adeda1ee36a99ef8b92d51920154/library/alloc/src/sync.rs#L1329
+        let _count = context.refcount.fetch_add(1, Ordering::Relaxed);
```

Comment:
Clone and drop are nearly the same as [Config](https://github.com/aws/s2n-tls/blob/main/bindings/rust/s2n-tls/src/config.rs#L124)